pkg/frontend/querymiddleware/shard_active_series.go

```diff
@@ -47,6 +57,17 @@ func newShardActiveSeriesMiddleware(upstream http.RoundTripper, limits Limits, l
 		limits:   limits,
 		logger:   logger,
 		encoder:  s2.NewWriter(nil),
+		bufferPool: sync.Pool{
+			New: func() any {
+				buf := make([]byte, jsoniterMaxBufferPoolSize)
+				return unsafe.SliceData(buf)
```

Comment:
Why are we doing this?

Reply:
See https://github.com/grafana/mimir/pull/7261#discussion_r1474287718